### DISCUSSION TITLE
Prior to upgrade to 3.4.x - Does Devices with same name can exists on same site but different location or rack ?

### DISCUSSION THREAD

**Discussion by Zeawiel** [netbox-community/netbox#11532] [Q&A]
Hello,

I'm looking to move from 3.3.5 to 3.4.x and I'm looking to Release note.
I noticed this breaking change:
> Device and virtual machine names are no longer case-sensitive. Attempting to create e.g. "device1" and "DEVICE1" within the same site will raise a validation error.

I run the reports I found in another discussion : https://github.com/netbox-community/reports/blob/master/reports/dcim-reports/case_insensitive_check.py
That's helped me a lot!

But I have some devices with the same name (same Case), withing the same site. 
But they are located in different Location and rack.
Does this is a blocking situation? (I expect it is).

Best regards

**Comment by candlerb** [comment]
Apart from case-sensitivity that logic hasn't changed between 3.3 and 3.4, so if it's accepted now it should still be accepted. (However it's not enforced by database constraints, it's enforced by Django models, so it's possible that data can get past it).

The logic is described [here](https://docs.netbox.dev/en/stable/models/dcim/device/):

> Device names must be unique within a site, unless the device has been assigned to a tenant. Devices may also be unnamed.

Do your devices currently have different tenants?

**Comment by Zeawiel** [comment]
Thank you for the answer.

Some have different tenants, but some have no tenant at all.
And indeed, when I tried to apply same tenant to two device with same name in different location:

> Errors
> 
>     Device with this Site, Tenant and Name already exists.

So it should work if : Different tenant or no tenant at all. 
I may need to think a way to make the name different. But it's bay devices (called 1/3 for example for card 1/3), I may need to add parent device named in the name of it, but it will be quite long. 
Hum.

**Reply by candlerb** [reply]
> So it should work if : Different tenant or no tenant at all.

It should work if one has a tenant and the other has no tenant. I would expect that if both have no tenant, then it will be rejected.

Naming of child devices is an interesting point, but in Netbox these are considered completely independent devices: the parent "device" is really just a dumb housing, a way of subdividing some rack space.  Child devices would need unique DNS names, so it seems reasonable to require them to have unique Netbox names.  This is intended for modelling blade servers, which are just like rack-mount servers but happen to fit inside an outer housing rather than directly into rack ears.

If you are using child devices to model line cards in a chassis switch or router, then you're Doing It Wrong™.  You should use either Modules in Module Bays (available in recent Netbox) or Inventory Items.  These are modular components of the same outer device, which share a single control plane, i.e. one management IP address.
